crates/bevy_text/src/font_atlas_set.rs

```diff
@@ -52,57 +52,58 @@ impl FontAtlasSet {
         font_size: f32,
         text: &str,
     ) -> f32 {
-        let font = fonts.get(&self.font).unwrap();
-        let scaled_font = ab_glyph::Font::as_scaled(&font.font, font_size);
-        let font_atlases = self
-            .font_atlases
-            .entry(FloatOrd(font_size))
-            .or_insert_with(|| {
-                vec![FontAtlas::new(
-                    textures,
-                    texture_atlases,
-                    Vec2::new(512.0, 512.0),
-                )]
-            });
-
-        let mut last_glyph: Option<Glyph> = None;
         let mut width = 0.0;
-        for character in text.chars() {
-            if character.is_control() {
-                continue;
-            }
-            let glyph = scaled_font.scaled_glyph(character);
-            if let Some(last_glyph) = last_glyph.take() {
-                width += scaled_font.kern(last_glyph.id, glyph.id);
-            }
-            if !font_atlases
-                .iter()
-                .any(|atlas| atlas.get_char_index(character).is_some())
-            {
-                if let Some(outlined_glyph) = scaled_font.outline_glyph(glyph.clone()) {
-                    let glyph_texture = Font::get_outlined_glyph_texture(outlined_glyph);
-                    let add_char_to_font_atlas = |atlas: &mut FontAtlas| -> bool {
-                        atlas.add_char(textures, texture_atlases, character, &glyph_texture)
-                    };
-                    if !font_atlases.iter_mut().any(add_char_to_font_atlas) {
-                        font_atlases.push(FontAtlas::new(
-                            textures,
-                            texture_atlases,
-                            Vec2::new(512.0, 512.0),
-                        ));
-                        if !font_atlases.last_mut().unwrap().add_char(
-                            textures,
-                            texture_atlases,
-                            character,
-                            &glyph_texture,
-                        ) {
-                            panic!("could not add character to newly created FontAtlas");
+        if let Some(font) = fonts.get(&self.font) {
```

Comment:
As an alternative, you could make an early return here in case the font doesn't exist yet. This way the rest of the function isn't indented.

Reply:
I think that would require this:
```rust
if fonts.get(&self.font).is_some() { return; }
let font = self.fonts.get(&self.font).unwrap();
```

Which is both a double hash, slightly less safe, and less idiomatic. I definitely love flattening out indentation, but im not sure thats the right call here.

Reply:
A simple way is:

```rust
let font = if let Some(font) = fonts.get(&self.font) { font } else { return };
```